### DISCUSSION TITLE
Disable keyboard interaction on ListBox

### DISCUSSION THREAD

**Discussion by SylarRuby** [tailwindlabs/headlessui#820] [Help]
Hi all. I've added a filter input inside the `ListBox` so that users can type to filter items. It will be common for them to type then enter "space" but [keyboard interaction](https://headlessui.dev/react/listbox#keyboard-interaction) is causing an issue:

```js
[..]

const filterOptions = (event) => {
  // This has no effect 😢
  event.stopPropagation()

  // code...
}

<Listbox.Options>

  <input
      value={query}
      onChange={filterOptions}
      placeholder="Filter..."
      type="text" 
    />

</Listbox.Options>

[..]

```

I'd like to prevent the dropdown from closing when the space bar is pressed.

**Comment by barbarah** [comment]
I have a similar setup. For me, changing the onKeyDown on the input works:

                    onKeyDown={(e) => { if (e.code === "Space") { e.stopPropagation(); } }}

This solution does feel a bit hacky. When searching for a solution I did find 3 discussions about the same issue. Non with better (or any) solutions. So there are multiple people running into the same issue with disabling keyboard interaction on space. Related discussions:

- https://github.com/tailwindlabs/headlessui/discussions/773
- https://github.com/tailwindlabs/headlessui/discussions/700
- https://github.com/tailwindlabs/headlessui/discussions/684

**Reply by StroverIT** [reply]
Thank you soo much!!!

**Reply by chekdotdev** [reply]
This is a great workaround. Thank you!

**Reply by akshay-rajput** [reply]
I'm using vue3 and add this on input worked for me `v-on:keydown.stop`

**Reply by elnur-elnur** [reply]
its worked in react.js . i put to input

**Reply by benjaminbarbe** [reply]
```
                  onKeyDown={(e) => { if (e.keyCode === 32) { e.stopPropagation(); } }}
```

**Reply by ADTC** [reply]
The same thing happens for [Menu's keyboard interaction](https://headlessui.com/react/menu#keyboard-interaction). Is this `onKeyDown` hack the only solution?

**Reply by ADTC** [reply]
@barbarah @benjaminbarbe it seems it's better to _always_ stop propagation, because typing a letter from `a-z` in the text box will focus an item that begins with the letter, and pressing `Space`/`Enter` will select it _while you're in the input text box_. (That should only happen outside the text box.)

```js
  onKeyDown={(e) => e.stopPropagation()} // prevent conflicts with Headless UI keyboard navigation
```

**Reply by venkat-spendflo** [reply]
Worked like a charm in react

**Reply by aeksco** [reply]
Here's my solution which retains the Headless UI keyboard interactions for navigating the listbox, but allows for an `<input>` element to to accept a lookup from a user:

```tsx
<input
    type="search"
    value={value}
    onChange={(e) => {
        onChange(e.currentTarget.value);
    }}
    onKeyDown={(e) => {
        const allowedHeadlessUIKeys = [
            "ArrowUp",
            "ArrowDown",
            "Enter",
            " ", // Space Key
            "Home",
            "End",
            "Escape",
        ];
        if (!allowedHeadlessUIKeys.includes(e.key)) {
            e.stopPropagation();
        }
    }}
/>
```

Works well enough for now, but would love to see official support for this use case!

**Comment by GreenImp** [comment]
Could you not just use the Combobox component instead? https://headlessui.com/react/combobox

It provides much the same functionality as the Listbox, but with the ability to search / filter results.

**Reply by chekdotdev** [reply]
@GreenImp `Combobox` works for interfaces where you want the "old" selection to go away as you're typing, but for interfaces where the current selection doesn't change until you select something new, it won't work.

Perfect example: GH labels: 
<img width="344" alt="Screenshot_20221117_154444" src="https://user-images.githubusercontent.com/28504852/202555926-1dbaec37-eae4-4617-92dd-0a4239d4a151.png">
